src/module.c

```diff
@@ -2764,7 +2837,58 @@ NAPI_METHOD(dcn_provider_get_status) {
   //TRACE("result %s", status);
 
   NAPI_RETURN_INT32(status)
-} 
+}
+
+// webxdc
+
+NAPI_METHOD(dcn_send_webxdc_status_update){
+  NAPI_ARGV(4);
+  NAPI_DCN_CONTEXT();
+  NAPI_ARGV_UINT32(msg_id, 1);
+  NAPI_ARGV_UTF8_MALLOC(json, 2);
+  NAPI_ARGV_UTF8_MALLOC(descr, 3);
+
+  int result = dc_send_webxdc_status_update(dcn_context->dc_context, msg_id, json, descr);
+  free(json);
+  free(descr);
+
+  NAPI_RETURN_UINT32(result);
+}
+
+NAPI_METHOD(dcn_get_webxdc_status_updates){
+  NAPI_ARGV(3);
+  NAPI_DCN_CONTEXT();
+  NAPI_ARGV_UINT32(msg_id, 1);
+  NAPI_ARGV_UINT32(status_update_id, 2);
+
+  char* result_json = dc_get_webxdc_status_updates(dcn_context->dc_context, msg_id, status_update_id);
+
+  NAPI_RETURN_AND_UNREF_STRING(result_json);
+}
+
+// todo dc_msg_get_webxdc_blob
+NAPI_METHOD(dcn_msg_get_webxdc_blob){
+  NAPI_ARGV(2);
+  NAPI_DC_MSG();
+  NAPI_ARGV_UTF8_MALLOC(filename, 1);
+
+  size_t size;
+  char* data = dc_msg_get_webxdc_blob(dc_msg, filename, &size);
+  free(filename);
+
+  // https://nodejs.org/api/n-api.html#napi_create_buffer_copy
+  napi_value jsbuffer;
+  NAPI_STATUS_THROWS(napi_create_buffer_copy(env,
+                                    size,
+                                    &data,
```

Comment:
Pass just the `data` here, not `&data`. `data` is already a pointer to data, and you pass a pointer to pointer here.